# This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:
#
# Copyright (c) 2025 Qualcomm Technologies, Inc. All rights reserved.
#
# SPDX-License-Identifier: BSD-3-Clause-Clear

From 4b56241bd9511fbe32405145bfb63cbdf944bf3a Mon Sep 17 00:00:00 2001
From: Ulrich Meis <umeis@qti.qualcomm.com>
Date: Fri, 2 Sep 2022 13:26:47 +0200
Subject: [PATCH 3/3] QcomModulePkg: KeyMaster: don't start twice.

In StartKeyMaster() check if we have already started it. Do nothing if
so. This happens to be the case when using 'fastboot boot'.

Change-Id: I89ec8ce035430cff67f031a09e28ef532c8f88bd
Signed-off-by: Ankurkumar Satasiya <asatasiy@qti.qualcomm.com>
---
 QcomModulePkg/Library/avb/KeymasterClient.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/QcomModulePkg/Library/avb/KeymasterClient.c b/QcomModulePkg/Library/avb/KeymasterClient.c
index 8b1d0941cc..c59cc38974 100644
--- a/QcomModulePkg/Library/avb/KeymasterClient.c
+++ b/QcomModulePkg/Library/avb/KeymasterClient.c
@@ -328,6 +328,8 @@ KeyMasterSetRotAndBootState (KMRotAndBootState *BootState)
   return Status;
 }
 
+static int startedKeyMaster = 0;
+
 EFI_STATUS
 StartKeyMaster (void)
 {
@@ -336,6 +338,12 @@ StartKeyMaster (void)
   KMSetBootStateReq BootStateReq = {0};
   KMSetBootStateRsp BootStateRsp = {0};
 
+  if (startedKeyMaster) {
+         return EFI_SUCCESS;
+  } else {
+         startedKeyMaster = 1;
+  }
+
   /*bring up keymaster*/
   EFI_STATUS Status = EFI_SUCCESS;
   KMHandle Handle = {NULL};
-- 
2.37.1

